### PR TITLE
Add unit tests documenting what happens when you call `createLiquidation()` with `maxTokensToLiquidate = 0`

### LINKED ISSUE #1671: Add unit tests for `createLiquidation` when `maxTokensToLiquidate = 0`

### PR DESCRIPTION
**Pull request by nicholaspai**
This unit test verifies this action as harmless. These are the takeaways:
- Dispute always succeeds because the `requiredCollateral = (tokensOutstanding * settlementPrice * collateralRequirement) = 0`. Even though the liquidated collateral is 0, 0 >= 0 in the check for `isDisputeSuccessful = Boolean(liquidatedCollateral >= requiredCollateral)`. 
- Because dispute succeeds, liquidator pays final fee. This is good behavior since the liquidator initiated a useless price request.
- The disputer can call `withdrawLiquidationRewards()` after the dispute resolves to get back their final fee bond.
- The liquidator and sponsor both cannot call `withdrawLiquidationRewards()` because the contract does not allow them to withdraw 0 rewards.
- The unfortunate side effect is that the liquidation data object can never be deleted since not all parties can "withdraw"

Resolves #1671 

Signed-off-by: Nick Pai <npai.nyc@gmail.com>

### CODE REVIEW

File: core/test/financial-templates/Liquidatable.js [hunk 1]
```diff
@@ -1231,6 +1231,89 @@ contract("Liquidatable", function(accounts) {
   });
 
   describe("Weird Edge cases", () => {
+    it("Liquidating 0 tokens always results in a successful Dispute", async () => {
+      // TODO: Liquidations for 0 tokens should be blocked but they are not. They will harmlessly always cause a successful dispute because
```
[the remark this answers is not in the document]
Comment (nicholaspai) [reply core/test/financial-templates/Liquidatable.js:1235]: I guess I left this TODO because we should really not even enable this situation in the first place. But we can remove this test when we block this behavior.

[the remark this answers is not in the document]
Comment (chrismaree) [reply core/test/financial-templates/Liquidatable.js]: cool. I saw you also updated issue #1553 so it's recorded there 👍
